src/queries/displayValue.ts

```diff
@@ -24,8 +25,13 @@ const getTextInputNodeByDisplayValue = (
   const { exact, normalizer } = options;
   const nodeValue =
     node.props.value !== undefined ? node.props.value : node.props.defaultValue;
+  const shouldReturnHostTextInput = getConfig().useBreakingChanges;
+  const typeToMatch = shouldReturnHostTextInput
+    ? getConfig().hostComponentNames?.textInput || 'TextInput'
```

Comment:
I wonder if we could do something better than hardcoding the default value here? 

Generally at this stage detectHostComponentNames should have been run. So perhaps we should throw an error if not? That should only happen in case when user did not use our `render` but rather bound functions to Test Renderer result using `within`. 

Or perhaps we could call `detectHostNameComponentsIfNeeded`. That would get the host component names even in such edge case scenario.

@pierrezimmermannbam wdyt?

Reply:
I hadn't found a proper way to deal with this but I just got an idea: what about we use `detectHostNameComponentsIfNeeded ` in the queries directly rather than on render ? That way we don't run autodetection for tests that don't need it and we make sure that it has been called when those queries are called. I just realized that `detectHostNameComponentsIfNeeded ` is called when using renderHook for instance which is something we don't want